pkg/pfconfig/semtechudp/semtechudp.go

```diff
@@ -24,8 +24,8 @@ import (
 
 // Config represents the full configuration for Semtech's UDP Packet Forwarder.
 type Config struct {
-	SX1301Conf  shared.SX1301Config `json:"SX1301_conf"`
-	GatewayConf GatewayConf         `json:"gateway_conf"`
+	SX1301Conf  []*shared.SX1301Config `json:"SX1301_conf"`
```

Comment:
This is not really backwards compatible, based on what I could find in the parsers for these files.

Could you please use a custom `MarshalJSON` that keeps the single object form (what was previously here) if there is only one config, and the array otherwise ?

Reply:
Added. Do we need an `UnmarshalJSON` as well?

Reply:
Yes, please.